### ISSUE TITLE
About the weighted_loss in experiment11

### ISSUE THREAD

**Issue by Kreutzer-C** [serdna1/ODIR-5K-multi-label-classification#1]
How do you calculate the weight of each class? I tried to use 1/(ni/N) as the weight of each class, where ni represents the number of each class and the N represents the total sample quantity. But got a terrible accuracy result on dev. data during training.

**Comment by serdna1** [comment]
Hi, @Kreutzer-C.

First of all, sorry for replying so late, i didn't see the issue before today.

The weights are calculated according to the offitial [BCEWithLogitsLoss](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html) documentation. There you can see this lines: 

> For example, if a dataset contains 100 positive and 300 negative examples of a single class, then pos_weight for the class should be equal to 300/100=3. The loss would act as if the dataset contains 3×100=300 positive examples.

I implemented a function called compute_loss_weights that returns a 1x8 tensor with a weight (total negatives / total positives) for each one of the 8 classes. You can find this function in the [utils script](https://github.com/serdna1/ODIR-5K-multi-label-classification/blob/main/scripts/utils.py). Then, the only thing i did is using the BCEWithLogits loss function with the parameter pos_weight set to the mentioned tensor. You can find this call in the [train script](https://github.com/serdna1/ODIR-5K-multi-label-classification/blob/main/scripts/train.py).

By the way, i think accuracy would be a bad metric for this unbalanced case. I use the mean of kappa, f1 and AUC as specified in the challenge.

**Comment by Kreutzer-C** [comment]
Thanks a lot! I will try again with the loss weights and use the new metrics that your mentioned ^_^
